src/common/tracing.ts

```diff
@@ -1,18 +1,147 @@
   autoInstrumentationsConfigMap: {
-    // eslint-disable-next-line @typescript-eslint/naming-convention
     '@opentelemetry/instrumentation-http': {
       ignoreIncomingRequestHook: (request): boolean =>
+        !(request.url?.startsWith(config.get('server.apiPrefix')) ?? false) ||
         IGNORED_INCOMING_TRACE_ROUTES.some((route) => request.url !== undefined && route.test(request.url)),
       ignoreOutgoingRequestHook: (request): boolean =>
         IGNORED_OUTGOING_TRACE_ROUTES.some((route) => typeof request.path === 'string' && route.test(request.path)),
     },
-    // eslint-disable-next-line @typescript-eslint/naming-convention
     '@opentelemetry/instrumentation-fs': {
       requireParentSpan: true,
+      endHook: (functionName, info): boolean => {
+        const key = fsFunctionMap.get(functionName);
+        if (key !== undefined && typeof info.args[0] === 'string') {
+          const masked = maskSchemasPath(info.args[0]);
+          if (masked) {
+            info.span.setAttribute(key, masked);
+          }
+        }
+        return true;
+      },
+      enabled: true,
+    },
+    '@opentelemetry/instrumentation-express': {
+      requestHook: (span, info): void => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        const operationId = info.request?.openapi?.schema?.operationId;
+        if (info.layerType === 'request_handler' && typeof operationId === 'string') {
+          span.updateName('request handler - ' + operationId);
+          span.setAttribute('http.openapi.operation-id', operationId);
+        }
+      },
     },
   },
 });
+/* eslint-enable @typescript-eslint/naming-convention */
+
+tracing.start();
+
+const tracer = trace.getTracer(SERVICE_NAME);
+
+export { tracing };
+
+export function callWithSpan<T>(fn: (span: Span) => T, spanName: string, spanOptions?: SpanOptions): T {
+  return tracer.startActiveSpan(spanName, spanOptions ?? {}, (span) => {
+    try {
+      const result = fn(span);
+      if (result instanceof Promise) {
+        // return new Promise<T>((resolve, reject) => {
+        result
+          .then(() => {
+            handleSpanOnSuccess(span);
+            // return resolve(r);
+          })
+          .catch((e) => {
+            handleSpanOnError(span, e);
+            // return reject(e);
+            // });
+          });
+        return result;
+      }
+      handleSpanOnSuccess(span);
+      return result;
+    } catch (error) {
+      handleSpanOnError(span, error);
+      throw error;
+    }
+  });
+}
+
+export function handleSpanOnSuccess(span: Span): void {
+  span.setStatus({ code: SpanStatusCode.OK });
+  span.end();
+}
+
+export function handleSpanOnError(span: Span, error?: unknown): void {
+  span.setStatus({ code: SpanStatusCode.ERROR });
+
+  if (error instanceof Error) {
+    span.recordException(error);
+  }
+
+  span.end();
+}
```

Comment:
As they pretty similar, consider combining them
```suggestion
export function handleSpan(span: Span, isSuccess: boolean, error?: unknown): void {
  span.setStatus({ code: isSuccess ? SpanStatusCode.OK : SpanStatusCode.ERROR });
  
  if (!isSuccess && error instanceof Error) {
    span.recordException(error);
  }
  
  span.end();
}
```

Reply:
as we talked i dont think we should do this for those reasons:
1. breaks the API set by telemetry package
2. less readable
3. adds two more ifs to each call